### PR TITLE
peer: make peer idle timeout a config var.

### LINKED ISSUE #383: config var for default connection timeout
"no answer for 5m0s -- disconnecting"
https://github.com/decred/dcrd/blob/master/peer/peer.go#L53

On an uplink that intermittently will drop all connections, 5m is really too long because thats easily the length of a whole block cycle, and a stake vote could get missed. I would like this to be a user config setting, and I would probably set it to 30 seconds for myself.

Or maybe it just makes sense to reduce the default const to 30 sec like the other const timeouts.

### PR DESCRIPTION
**Pull request by jcmincke**
Fixes #383 

Please, see my comments in https://github.com/decred/dcrd/issues/383
before reviewing.

### CODE REVIEW

**Comment by davecgh** [comment]
Thanks for the PR.  As an FYI, we're currently working on getting the v1.0.0 release out, so I've put this in the v1.1.1 milestone.  It will probably be a few days before this receives proper review.

**Comment by dajohi** [comment]
Tests not passing...

File: config.go [hunk 1]
```diff
@@ -160,6 +161,7 @@ type config struct {
 	PipeRx               uint          `long:"piperx" description:"File descriptor of read end pipe to enable parent -> child process communication"`
 	PipeTx               uint          `long:"pipetx" description:"File descriptor of write end pipe to enable parent <- child process communication"`
 	LifetimeEvents       bool          `long:"lifetimeevents" description:"Send lifetime notifications over the TX pipe"`
+	PeerIdleTimeout      time.Duration `long:"peerideltimeout" description:"Duration of inactivity before a peer is timed out"`
```
Comment (dajohi) [inline config.go]: spelling...

File: glide.lock [hunk 2]
```diff
@@ -1,8 +1,8 @@
 hash: 5ff55cbde58db9eecfef0a796b2513f910996d6aaa1244f31475852835e48efe
-updated: 2017-08-30T23:46:58.835362449Z
```
Comment (dajohi) [inline glide.lock]: glide.lock should not be updated.

File: peer/peer.go [hunk 3]
```diff
@@ -2048,6 +2048,7 @@ func NewInboundPeer(cfg *Config) *Peer {
 
 // NewOutboundPeer returns a new outbound decred peer.
 func NewOutboundPeer(cfg *Config, addr string) (*Peer, error) {
+
```
Comment (dajohi) [inline peer/peer.go]: unneeded.

Comment (jcmincke) [reply peer/peer.go]: Sorry, what is unneeded?

Comment (dajohi) [reply peer/peer.go]: added empty line

File: peer/peer.go [hunk 4]
```diff
@@ -243,6 +240,9 @@ type Config struct {
 	// Listeners houses callback functions to be invoked on receiving peer
 	// messages.
 	Listeners MessageListeners
+
+	// idleTimeout is the duration of inactivity before we time out a peer.
```
Comment (chappjc) [inline peer/peer.go]: // IdleTimeout

**Comment by jcmincke** [comment]
please close this PR:

It is replaced by https://github.com/decred/dcrd/pull/856

**Comment by jolan** [comment]
Closing per submitter's request in favor of #856.

**Pull request closed without merging**